scripts/hotspot.js

```diff
@@ -13,20 +13,25 @@
    * @param  {number} id
    * @param  {boolean} isSmallDeviceCB
    * @param  {H5P.ImageHotspots} parent
+   * @param  {string} icon
+   * @param  {image} iconImage
    */
-  ImageHotspots.Hotspot = function (config, color, id, isSmallDeviceCB, parent) {
+  ImageHotspots.Hotspot = function (config, color, id, isSmallDeviceCB, parent, icon, iconImage) {
     var self = this;
     this.config = config;
     this.visible = false;
     this.id = id;
     this.isSmallDeviceCB = isSmallDeviceCB;
+    // A utility variable so you don't have to always check if there is an iconImage.
+    var iconImageExists = (iconImage == null ? false : true); 
```

Comment:
I ran into a bug when an image is uploaded, but the Predefined Icon is selected it will still show the image. To resolve this, I'm using a if/else to set iconImageExists. https://github.com/nmillin/h5p-image-hotspots/commit/efa5ed243b26df39bd46b076c8417a299d527c3d

This has resolved the bug, but let me know if you have other thoughts on this.